controllers/storagecluster/uninstall_reconciler.go

```diff
@@ -270,15 +272,28 @@ func (r *StorageClusterReconciler) reconcileUninstallAnnotations(sc *ocsv1.Stora
 	return false, nil
 }
 
-// verifyNoStorageConsumerExist verifies there are no storageConsumers on the same namespace
-func (r *StorageClusterReconciler) verifyNoStorageConsumerExist(instance *ocsv1.StorageCluster) error {
+// verifyNoStorageConsumerExist verifies there are no storageConsumers on the same namespace conditionally
+func (r *StorageClusterReconciler) verifyNoStorageConsumerExist(instance *ocsv1.StorageCluster, skipLocalConsumer bool) error {
 
 	storageConsumers := &ocsv1alpha1.StorageConsumerList{}
 	err := r.Client.List(context.TODO(), storageConsumers, &client.ListOptions{Namespace: instance.Namespace})
 	if err != nil {
 		return err
 	}
 
+	if len(storageConsumers.Items) == 1 && skipLocalConsumer {
+		clusterID := util.GetClusterID(r.ctx, r.Client, &r.Log)
+		if clusterID == "" {
+			return fmt.Errorf("failed to get cluster id")
+		}
+
+		consumer := &storageConsumers.Items[0]
+		if strings.HasSuffix(consumer.Name, clusterID) {
+			r.Log.Info("Only local storage consumer exist")
+			return nil
+		}
+	}
```

Comment:
I think this is the wrong place and the wrong assumption. From a flow perspective, we want:
1. delete local storage client
2. check if we have any storage clients (should be 0) or fail reconciliation 
3. delete storage Provider.

Right now this is hard to implement because of the whole `resourceManager` design which limits us. 
With that in mind, I would suggest having the check inside the ensure deleted of the API server. This should work because the entire system is eventually consistent

Reply:
this check is here for noobaa to go into deletion mode, w/o skipping local client we don't make progress in uninstallation.

Reply:
But why would noobaa fo into deletion mode if we exit the reconcile beforehand

Reply:
I mean we want noobaa to go into deletion mode if we are uninstalling storagecluster and there's only local storageconsumer.

Before this PR, uninstallation of storagecluster can only happen if there are no storageconsumers, however that'll not work as local storageconsumer can't be deleted due to noobaa and noobaa can't be marked for deletion due to storageconsumer existence.

Reply:
I don't see the ordering problem here:
Creation/reconciliation  order is: 
1. ProviderServer
2. StorageClient
3. NooBaa 

And the reverse order for deletion/removal

Reply:
done.